mono/mini/method-to-ir.c

```diff
@@ -9107,21 +9107,8 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 					ins->inst_p0 = cmethod;
 					ins->inst_p1 = arg_array [0];
 					MONO_ADD_INS (cfg->cbb, ins);
-					link_bblock (cfg, cfg->cbb, end_bblock);
-					start_new_bblock = 1;
-
-					// FIXME: Eliminate unreachable epilogs
-
-					/*
-					 * OP_TAILCALL has no return value, so skip the CEE_RET if it is
-					 * only reachable from this call.
-					 */
-					GET_BBLOCK (cfg, tblock, ip + 5);
-					if (tblock == cfg->cbb || tblock->in_count == 0)
-						skip_ret = TRUE;
-					push_res = FALSE;
 
-					goto call_end;
+					goto tailcall_remove_ret;
```

Comment:
This is jumping to the taken branch of an `if` statement. How feasible is to move all the code for CEE_CALL / CEE_CALLVIRT to a separate file and properly factor into a saner set of functions?

Reply:
Almost anything is possible. I’d rather take this change in the short term & refactor separately/later.  

Placing a label here and jmping to is not a problem & better than duplicating code. 

Goto is not actually evil despite what some people say.